test/proxy.js

```diff
@@ -9,18 +9,79 @@
  * tests less flaky due to changes in the API or API data.
  */
 const process = require('process')
+const zlib = require('zlib')
 
 const talkback = require('talkback')
 
 const host = 'https://api.openverse.engineering/v1'
 
 const tapeNameGenerator = (tapeNumber) => `response-${tapeNumber}`
 
-const updatingTapes = process.argv.includes('--update-tapes')
+const updatingTapes =
+  process.argv.includes('--update-tapes') || process.env.UPDATE_TAPES === 'true'
+
 const recordMode = updatingTapes
   ? talkback.Options.RecordMode.NEW
   : talkback.Options.RecordMode.DISABLED
 console.log('Record mode: ', recordMode)
+
+/**
+ * Transform any response values to use the talkback
+ * proxy instead of pointing directly upstream for
+ * RESTful references.
+ *
+ * Ignore thumbnail and non-successful requests, those
+ * don't return JSON bodies so we can save them as-is.
+ *
+ * Sometimes we get a raw brotli buffer back from upstream
+ * and we need to decompress it. Sometimes we get the actual
+ * JSON. I'm fairly confident this has something to do with Cloudflare
+ * cached responses being compressed and others not? In any case,
+ * if the response comes back compressed then we need to
+ * decompress it, fix the upstream API references, and then
+ * compress it back before saving. We could mess with the
+ * `content-encoding` header instead of re-compressing it but
+ * I think that would sort of violate the contract talkback is meant
+ * to have and furthermore would eliminate a complexity in
+ * our stack that apparently exists in production. Given e2e
+ * tests should aim to test as close to production conditions as
+ * possible I think it makes sense to retain both the compressed
+ * and uncompressed responses.
+ *
+ * A note to future contributors: If you find that JSON.parse is
+ * complaining about unknown characters then you're either
+ * dealing with an error response that isn't being caught by
+ * the status check on the first line OR you've discovered
+ * another compression algorithm being used (or, of course, it
+ * could be something else entirely). Don't discount that it could
+ * be something else, but I'd check those two things first
+ * before digging elsewhere.
+ *
```

Comment:
Your approach is likely better and more robust, anyway, but I believe we could also bypass cache hits entirely by adding a random query param to the URL.

Reply:
Ah we could! Good to keep in mind for other types of testing. But I do think maintaining an API proxy that mirrors production as close as possible is beneficial. Normal requests aren't sending random query params after all.